File: APG4bPython/ex14.py
data = [int(c) for c in input().split()]
# dataの中で隣り合う等しい要素が存在するなら"YES"を出力し、そうでなければ"NO"を出力する
# ここにプログラムを追記
count = 0

for i in range(len(data)-1):
    if data[i+1] == data[i]:
        print("YES")
        break
    else:
        count += 1
        if count == len(data)-1:
            print("NO")

File: Beginner/APG4bRust/src/bin/ex14.rs
use proconio::input;

fn main() {
    input! {
        a: [i64;5],
    }

    if a[0] == a[1] || a[3] == a[4] {
        println!("YES");
        return
    }

    for i in 1..3 {
        if a[i] == a[i-1] || a[i] == a[i+1] {
            println!("YES");
            return
        }
    }

    println!("NO");
}